### ISSUE TITLE
The emulator behaves differently than the production version with respect to the rawBody property

### ISSUE THREAD

**Issue by grimmen** [GoogleCloudPlatform/cloud-functions-emulator#161]
The rawBody property of the request - it is present when I run the function deployed to the cloud, but not when I run the function locally using "firebase serve --only functions".  I have tried to install @latest of both firebase-tools and firebase.

#### Environment details

  - OS: OSX
  - Node.js version: 6.11.1
  - npm version:
  - @google-cloud/functions-emulator version:  firebase-functions@0.7.2

#### Steps to reproduce

  1. Receive request with body content.
  2. Check if request.rawBody property is defined and contains the byte of the body, log result
  3. Compare running the function in emulation vs. in prod

**Comment by justinwh** [comment]
I ran into this issue as well (following [these docs](https://cloud.google.com/functions/docs/writing/http#handling_file_uploads)) and at least as of `1.0.0-beta.4` I have code working in both production and the emulator with the following workaround:

Replace the call to `busboy.end(req.rawBody)` with...

```
if (req.rawBody) {
    busboy.end(req.rawBody);
}
else {
    req.pipe(busboy);
}
```

It pipes the request body in the emulator, where `req.rawBody` doesn't exist, and uses the documented behavior in production.

**Comment by robertlevy** [comment]
This should really be fixed.  The sample code @ https://cloud.google.com/functions/docs/writing/http refers req.pipe which only works in the emulator, not when deployed.

**Comment by jmdobry** [comment]
This _should_ be fixed in [1.0.0-beta.5](https://github.com/GoogleCloudPlatform/cloud-functions-emulator/releases/tag/1.0.0-beta.5)
